apps/aecore/src/aec_mining.erl

```diff
@@ -38,18 +38,8 @@ need_to_regenerate(Block) ->
     (MaxTxsInBlockCount =/= CurrentTxsBlockCount)
         andalso (lists:sort(get_txs_to_mine_in_pool()) =/= lists:sort(Txs)).
 
--spec mine(aec_blocks:block(), aec_pow:nonce()) -> {ok, aec_blocks:block()} | {error, term()}.
-mine(Block, Nonce) ->
-    Target = aec_blocks:target(Block),
-    BlockBin = aec_headers:serialize_for_hash(aec_blocks:to_header(Block)),
-    case aec_pow_cuckoo:generate(BlockBin, Target, Nonce) of
-        {ok, {Nonce, Evd}} ->
-            {ok, aec_blocks:set_pow(Block, Nonce, Evd)};
-        {error, no_solution} = Error ->
-            Error;
-        {error, {runtime, _}} = Error ->
-            Error
-    end.
+mine(BlockBin, Target, Nonce) ->
```

Comment:
Variable name without computation visible may confuse reader, I suggest renaming `BlockBin` to `HeaderBin`. Though totally optional of course.

Reply:
Agreed. This is a better name. I should also put the spec back.